### ISSUE TITLE
Cannot read property 'latitude' of undefined

### ISSUE THREAD

**Issue by kursata** [manuelbieh/geolib#82]
Hi 

We're experiencing following error while calling findNearest() function. The error occurs in some of the calls. The inputs parameters of successful and unsuccessful calls are identical. 
Any idea on the cause of the error or a fix on handling the type error will be appreciated.  

TypeError: Cannot read property 'latitude' of undefined
    at Geolib.geolib.extend.coords (/var/app/current/node_modules/geolib/dist/geolib.js:179:121)
    at Geolib.geolib.extend.getDistance (/var/app/current/node_modules/geolib/dist/geolib.js:248:26)
    at Geolib.geolib.extend.orderByDistance (/var/app/current/node_modules/geolib/dist/geolib.js:924:37)
    at Geolib.geolib.extend.findNearest (/var/app/current/node_modules/geolib/dist/geolib.js:1013:32)
    at /var/app/current/BAL/Content/Watchlist/Watchlist.js:633:56
    at fn (/var/app/current/node_modules/async/lib/async.js:746:34)
    at /var/app/current/node_modules/async/lib/async.js:1213:16
    at /var/app/current/node_modules/async/lib/async.js:166:37
    at /var/app/current/node_modules/async/lib/async.js:706:43
    at /var/app/current/node_modules/async/lib/async.js:167:37
    at Immediate._onImmediate (/var/app/current/node_modules/async/lib/async.js:1206:34)
    at processImmediate [as _immediateCallback](timers.js:383:17)

**Comment by lprhodes** [comment]
Seeing the exact same issue.

**Comment by manuelbieh** [comment]
Can you provide some code examples maybe?

**Comment by lprhodes** [comment]
I'm debugging right the issue at the moment - I'll update shortly.

**Comment by lprhodes** [comment]
@manuelbieh It occurs when coordinates with additional precision are provided. 

e.g. { latitude: 40.7536853999999, longitude: -73.9991637 }

**Comment by nitindhar7** [comment]
@manuelbieh might it be related to this #113 ?

**Comment by neilakoh** [comment]
I am also having this issue right now but only on IOS. . it works fine on android . . in my case I added onPress event on the map to get coordinates. Here is my code:

https://github.com/airbnb/react-native-maps/issues/1499

It appears that on IOS the callback is missing some data or it did not return properly. Please advise. Thank you.

**Comment by rayzor65** [comment]
I believe I am having a similar issue
`                        <MapView.Marker
                            key={place.placeId}
                            coordinate={place.latLng}
                            title={place.title}
                            description={place.address}
                            onPress={this.animateToCoordinate}
                        >
                            <MapView.Callout>
                                <PlaceCallout {...place} />
                            </MapView.Callout>
                        </MapView.Marker>

    animateToCoordinate(e) {
        console.log(e);
        console.log(e.nativeEvent);
        console.log(e.nativeEvent.coordinate);
    }`

**Comment by neilakoh** [comment]
any update on this?

**Comment by ianawilson** [comment]
I don't know if this is the same issues others are seeing, but I also ran into it like this:

This (and #130) can both be caused by using the `navigator.geolocation`APIs with geolib. The [`Coordinates`](https://developer.mozilla.org/en-US/docs/Web/API/Coordinates) object returned by a call to `getCurrentPosition()` or `watchPosition()` doesn't `hasOwnProperty()` for either `'latitude'` or `'longitude'`, although the implementing object can retrieve values for these properties.

This means that [`coords()`, delegating to `getKeys()`'s, fails to find latitude and longitude properties](https://github.com/manuelbieh/Geolib/blob/4fed5b6803d8eddb70ae488e9258b6770f2c9a7a/src/geolib.js#L110), and returns `undefined`. Other functions that depend on `coords()` (most of them, I believe), throw this error because they assume that `coords()` will return a valid object, but it does not, resulting in this error.

**Comment by kirakik** [comment]
@ianawilson did you figure out how to fix this?

**Comment by ianawilson** [comment]
@kirakik My workaround was to map the `Coordinates` object to a POJO, eg,

```js
navigator.geolocation.getCurrentPosition(function(position) {
  // or whatever geolib function
  geolib.longitude({
    latitude: position.coords.latitude,
    longitude: position.coords.longitude,
  });
});
```

or map when I had a collection of positions,

```js
geolib.getPathLength(positions.map((p) => ({
  latitude: p.coords.latitude,
  longitude: p.coords.longitude,
})));
```

But I would love to see geolib work nicer with the geolocation API!

**Comment by manuelbieh** [comment]
I was pretty sure that already worked 🤔  Should be fixed again in v3.0.0 which I will release shortly.
